rticonnextdds_connector/rticonnextdds_connector.py

```diff
@@ -338,12 +353,41 @@ def __init__(self, input, index = -1):
 	def valid_data(self):
 		"""Returns whether this sample contains valid data
 
-		If this returns ``False``, the data getters (``get_dictionary()``, ``get_number()``...)
-		cannot be called.
+		If this returns ``False``, this object's getters cannot be called.
 		"""
 
 		return self.input.infos.isValid(self.index)
 
+	def __getitem__(self, field_name):
+		number_value = ctypes.c_double()
+		bool_value = ctypes.c_int()
+		string_value = ctypes.c_char_p()
+		selection = ctypes.c_int()
+		retcode = rtin_RTIDDSConnector_getAnyValueFromSamples(
+			self.input.connector.native,
+			ctypes.byref(number_value),
+			ctypes.byref(bool_value),
+			ctypes.byref(string_value),
+			ctypes.byref(selection),
+			tocstring(self.input.name),
+			self.index + 1,
+			tocstring(field_name))
+		_check_retcode(retcode)
+
+		if retcode == _ReturnCode.no_data:
+			return None
+
+		if selection.value == 1:
+			return number_value.value
+		elif selection.value == 2:
+			return bool_value.value
+		elif selection.value == 3:
+			python_str_value = fromcstring(cast(string_value, c_char_p).value)
```

Comment:
Should we make these two line (fromcstring and freeString) into a simple helper function for us? It seems like something we will have to do every time we deal with C strings